--- a/linux/CMakeLists.txt
+++ b/linux/CMakeLists.txt
@@ -60,10 +60,17 @@
     set(OBJECTBOX_ARCH armv6hf)
 endif ()
 
+if (${OBJECTBOX_ARCH} STREQUAL "x64")
+    set(OBJECTBOX_HASH d1a22f5a43e8aa438c987524c2ef97f5d179acb991b0f63ab03a30c33b882368)
+elseif (${OBJECTBOX_ARCH} STREQUAL "aarch64")
+    set(OBJECTBOX_HASH 948fe456904b8b0ae6b22a26a9f460734dc3c2ec58750df3a329ccae1ebe035d)
+endif ()
+
 include(FetchContent)
 FetchContent_Declare(
     objectbox-download
     URL https://github.com/objectbox/objectbox-c/releases/download/v${OBJECTBOX_VERSION}/objectbox-linux-${OBJECTBOX_ARCH}.tar.gz
+    URL_HASH SHA256=${OBJECTBOX_HASH}
 )
 
 FetchContent_GetProperties(objectbox-download)
